File: src/temp/pulsar/Commands.kt
package temp.pulsar

import kotlin.system.exitProcess

fun help() {
    println("Usage: pulsar [command] [file]")
    println("Commands:")
    println("    -h     : Shows This Help Menu")
    println("    -v     : Displays The Current Version Number")
    println("    -i     : Interprets The Given File")
    println("    -c     : Compiles The Given File")
    println("    -d     : Displays Debug Information\n" +
            "             And Interprets The File\n")
}

fun error(message: String) {
    help()
    println(message)
    exitProcess(1)
}

fun parseCommands(command: String) {
    when (command) {
        "-h" -> help()
        "-v" -> version()
        else -> error("Invalid Command For One Argument: $command")
    }
}

fun parseCommands(command: String, file: String) {
    setUp(file)
    when (command) {
        "-h" -> help()
        "-i" -> Pulsar.interpretFile(file)
        "-c" -> Pulsar.compileFile(file)

        "-d" -> {
            debug()
            Pulsar.interpretFile(file)
        }

        else -> error("Invalid Command For Two Arguments: $command")
    }
}

// Set To Debug Mode
fun debug() {
    Pulsar.conditions.debug = true
}

// Displays The Current Version (Currently Hardcoded In Pulsar.java)
fun version() {
    println("Version: ${Pulsar.conditions.version}")
}

// Store The Name Of The Input File For Future Use
fun setUp(name: String) {
    Pulsar.conditions.fileIn = name.substring(0, name.lastIndexOf("."))
}

File: src/codepulsar/lang/Commands.cpp
#include "Commands.h"


void Pulsar::help() {
    std::cout << "Usage: pulsar [command] [file]" << std::endl;
    std::cout << "Commands:" << std::endl;
    std::cout << "    -h     : Shows This Help Menu" << std::endl;
    std::cout << "    -v     : Displays The Current Version Number" << std::endl;
    std::cout << "    -i     : Interprets The Given File" << std::endl;
    std::cout << "    -c     : Compiles The Given File" << std::endl;
    std::cout << "    -d     : Displays Debug Information\n"
              << "             And Interprets The File\n" << std::endl;
}

void Pulsar::error(std::string message) {
    help();
    std::cout << "Set Up Error | " << message << std::endl;
    exit(1);
}

void Pulsar::parseCommands(std::string command) {
    if (command == "-h") help();
    else if (command == "-v") version();
    else error("Invalid One Argument Command: " + command);
}

void Pulsar::parseCommands(std::string command, std::string file) {
    setUp(file);

    if (command == "-h") help();
    else if (command == "-i") interpretFile(file);
    else if (command == "-c") compileFile(file);

    else if (command == "-d") {
        debug();
        interpretFile(file);
    }

    else error("Invalid Two Argument Command: " + command);
}

void Pulsar::debug() {
    conditions.debug = true;
}

void Pulsar::version() {
    std::cout << "Version: " << conditions.version << std::endl;
}

void Pulsar::setUp(std::string name) {
    conditions.fileIn = name.substr(0, name.find_last_of('.'));
}
